src/parse.c

```diff
@@ -4865,7 +4869,29 @@ static tree_t p_interface_function_specification(void)
    }
 
    consume(tRETURN);
-   type_set_result(type, p_type_mark());
+   if (standard() < STD_19) {
+      if (peek_nth(2) == tOF)
+         require_std(STD_19, "function knows return type");
+      type_set_result(type, p_type_mark());
+   } else {
+      if (peek_nth(2) == tOF) {
+         ident_t id = p_identifier();
+
+         tree_t t = tree_new(T_SUBTYPE_DECL);
```

Comment:
Actually the declaration object should only go in the function body. For the specification we just need to create a new subtype `type_t` (I forgot to explain that bit). So something like:

```c
type_t sub = type_new(T_SUBTYPE);
type_set_id(sub, p_identifier());
type_set_base(sub, p_type_mark());

type_set_result(type, sub);
```

Reply:
I don't quite understand what you mean by this.  We want to create a new `T_SUBTYPE` or `T_SUBTYPE_DECL`?

Reply:
> I don't quite understand what you mean by this. We want to create a new `T_SUBTYPE` or `T_SUBTYPE_DECL`?

Both. The `type_t` needs to get created in the specification and used as the result type, the `tree_t` needs to get created in the body and added to the declaration list.